pcdsdevices/interface.py

```diff
@@ -227,8 +227,13 @@ def __init_subclass__(cls, **kwargs):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self._tab = self._class_tab.new_instance(self)
+        self._tab_initialized = False
 
     def __dir__(self):
+        if not self._tab_initialized and hasattr(self, 'presets'):
+            self._tab_initialized = True
+            self.presets.sync()
```

Comment:
Change/addition request: we need to identify other times where presets may need a late sync. For example, if someone is expecting presets to be available in a hutch-python function call they need a clear way to trigger this without tabbing (or we need to load them automatically).

Reply:
The `device.presets.sync()` call is still available, and seems to be how we suggest people sync presets across sections.  I imagine we haven't advertised this as thoroughly as necessary.

Reply:
I guess we could overwrite `__getattribute__` and catch attempts to access `mv_*`, `umv_*`, and `wm_*` methods, but are there any other access modes we can think of?

Reply:
Is it possible that doing this on `__getattribute__` is actually most correct to make sure the loaded values are always up-to-date?

Reply:
My one qualm with this is that it does break the already fragile composition abstraction barrier we've set up with the Presets module.  Maybe there's also a way around this.  

Back to the mines!